core/src/main/scala/com/databricks/labs/remorph/intermediate/adf/activities/Activity.scala

```diff
@@ -0,0 +1,29 @@
+package com.databricks.labs.remorph.intermediate.adf.activities
+
+import com.databricks.labs.remorph.intermediate.adf.PipelineNode
+import com.databricks.labs.remorph.intermediate.adf.services.LinkedServiceReference
+
+case class Activity(
+    dependsOn: Seq[ActivityDependency],
+    description: Option[String],
+    linkedServiceName: Option[LinkedServiceReference],
+    name: Option[String],
+    onInactiveMarkAs: Option[OnInactiveMarkAs],
+    policy: Option[ActivityPolicy],
+    state: Option[ActivityState],
+    activityType: Option[String],
+    activityProperties: Option[ActivityProperties],
+    userProperties: Seq[UserProperty])
```

Comment:
Case classes like this one where all fields are `Seq` or `Option` almost always indicate a design problem. As a rule of thumb, one should ask themselves "what would be the meaning of `Activity(Seq(), None, None, None, None, None, None, None, None, Seq())`?".

`Option` in a case class field should only be used when there is a meaningful situation where the field could be absent.

For example, is it possible for an `Activity` to have no `state`? Note that isn't exactly the same question as "can an `Activity` be neither `Active` nor `Inactive`?". If the answer to the latter question is "yes", then we should add a `case object Unknown extends Activity` for example and define the field as `state: ActivityState = Unknown`.

Reply:
I've made the `name`, `activityType`, `activityProperties`, and `state` fields required. Every activity should have at minimum a name, type, set of type-specific properties, and state.

Many of the other fields are configurations which will be optionally returned by the API client.